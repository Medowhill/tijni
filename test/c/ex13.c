#include <stdlib.h>

int *f() {
  return (int *) malloc(sizeof(int));
}

int *g() {
  return (int *) malloc(sizeof(int));
}

int *h(int x) {
  int *p = f();
  int *q = g();
  int *r = p;
  if (x) r = q;
  *r = x;
  return r;
}
