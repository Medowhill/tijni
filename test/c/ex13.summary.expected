{
  "f": {
    "params": [],
    "ret": [ "Allocsite(0)" ],
    "env": {
      "%call(f)": [ "Allocsite(0)" ],
      "%0(f)": [ "Allocsite(0)" ]
    }
  },
  "g": {
    "params": [],
    "ret": [ "Allocsite(1)" ],
    "env": {
      "%call(g)": [ "Allocsite(1)" ],
      "%0(g)": [ "Allocsite(1)" ]
    }
  },
  "h": {
    "params": [ { "name": "%x(h)", "used": false, "freed": false } ],
    "ret": [ "Allocsite(0)", "Allocsite(1)" ],
    "env": {
      "%call(f)": [ "Allocsite(0)" ],
      "%0(f)": [ "Allocsite(0)" ],
      "%call(g)": [ "Allocsite(1)" ],
      "%0(g)": [ "Allocsite(1)" ],
      "%x(h)": [ "'%x(h)" ],
      "%call(h)": [ "Allocsite(0)" ],
      "%call1(h)": [ "Allocsite(1)" ],
      "%tobool(h)": [],
      "%r.0(h)": [ "Allocsite(0)", "Allocsite(1)" ],
      "Allocsite(0)": [ "'%x(h)" ],
      "Allocsite(1)": [ "'%x(h)" ]
    }
  }
}